crates/collator-selection/src/lib.rs

```diff
@@ -427,6 +430,7 @@ pub mod pallet {
         /// Kicks out candidates that did not produce a block in the kick threshold
         /// and refund their deposits.
         pub fn kick_stale_candidates(candidates: Vec<CandidateInfo<T::AccountId, BalanceOf<T>>>) -> Vec<T::AccountId> {
+            // TODO: also kick candidates when escrow balance drops
```

Comment:
I don't think there is consensus on whether or not we should, so fine with leaving this as-is for now